applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_implicit_dynamic_solver.py

```diff
@@ -108,3 +174,22 @@ def _bdf_integration_order(self):
             KratosMultiphysics.Logger.PrintWarning("WARNING:: BDF Order: ", str(order) + " constant time step must be considered")
 
         return order
+
+
+    def __extract_model_parts(self, params_list):
+        """Extracting a list of SubModelParts to be added to the ComputingModelPart
+        If the MainModelPart is to be added to the ComputingModelPart, then this is
+        done directly, no need to also add the SubModelParts, since they are contained
+        in the MainModelpart
+        """
+        list_model_part_names = [params_list[i].GetString() for i in range(params_list.size())]
+        main_model_part_name = self.settings["model_part_name"].GetString()
+        model_parts_list = []
+        for model_part_name  in list_model_part_names:
+            # only SubModelParts of the MainModelPart can be used!
+            if model_part_name != main_model_part_name:
+                full_model_part_name = main_model_part_name + "." + model_part_name
+            else:
+                full_model_part_name = main_model_part_name
+            model_parts_list.append(self.model[full_model_part_name])
+        return model_parts_list
```

Comment:
Isn't this implementation the same as in the check-and-prepare-process? => please refactor to an external file and not copy-paste
These changes are deliberate